src/include/daos/object.h

```diff
@@ -407,4 +407,74 @@ struct obj_enum_rec {
 	uint32_t		rec_flags;
 };
 
+enum daos_recx_type {
+	/** normal valid recx */
+	DRT_NORMAL	= 0,
+	/** hole recx */
+	DRT_HOLE	= 1,
+	/**
+	 * shadow valid recx, only used for EC degraded fetch to indicate
+	 * recx on shadow, i.e need-to-be-recovered recx.
+	 */
+	DRT_SHADOW	= 2,
+};
+
+struct daos_recx_ep {
+	daos_recx_t		re_recx;
+	daos_epoch_t		re_ep;
+	enum daos_recx_type	re_type;
+};
+
+struct daos_recx_ep_list {
+	/** #valid items in re_items array */
+	uint32_t		 re_nr;
+	/** #total items (capacity) in re_items array */
+	uint32_t		 re_total;
+	/** epoch valid flag, re_items' re_ep can be ignored when it is false */
+	uint32_t		 re_ep_valid:1;
+	struct daos_recx_ep	*re_items;
+};
```

Comment:
the bit flag is to differentiate if the data epoch is valid - if only use for io map then  can ignore the epoch when pack to RPC.

Reply:
we normally use bit-flag only if it can share with other bytes and save a few bytes, otherwise we might just use bool. e.g. 
```
 uint32_t   foo:31:
 uint32_t    bar:1;
```
this only takes 4 bytes, but if you remove 31 after foo, it takes 8 bytes and it's better just use bool